### ISSUE TITLE
Adding events minimized, maximized and restored

### ISSUE THREAD

**Issue by BillBridge** [r0x0r/pywebview#828]
### Specification
- pywebview version: 3.5
- operating system: Windows 10
- web renderer: 

### Description  

Would it be possibe to add the following events:
window.minimized      # Event that is fired when pywebview window is minimized
window.maximized     # Event that is fired when pywebview window is maximized
window.restored         # Event that is fired when pywebview window is restored
  
### Practicalities

- NO I am willing to work on this issue myself.

- YES I am prepared to support this issue financially.

**Comment by r0x0r** [comment]
Yes, this could be done. If you are prepared to sponsor this, I can implement this.

The estimated workload for this feature is 5h at 70€/h.

**Comment by BillBridge** [comment]
Hi r0x0r,
Apologies for the late reply. 
Yes I am prepared to sponsor this for 5h at 70 euro's.
Please inform me how to proceed on this.
Thanks very much for your cooperation. :)

**Comment by BillBridge** [comment]
> Yes, this could be done. If you are prepared to sponsor this, I can implement this.
> 
> The estimated workload for this feature is 5h at 70€/h.

Dear Roman,
Yes I am prepared to sponsor this. Please inform me how to proceed on this.
Thanks very much for your reply.

**Comment by r0x0r** [comment]
@BillBridge Got it. I will get to work on this. Will make an expense claim via Open Collective.

**Comment by BillBridge** [comment]
> @BillBridge Got it. I will get to work on this. Will make an expense claim via Open Collective.

Great Roman,
Thanks very much.
Looking forward to your expense claim.

**Comment by r0x0r** [comment]
@BillBridge
PR is here https://github.com/r0x0r/pywebview/pull/842 . I ran into an unexpected problem, namely that there is already a `minimized` property on the window object, so minimized as the event name could not be used. After much thought, I decided to migrate events to a new naming convention: `on_minimized`, `on_maximized`, `on_restored`. New events follow this naming scheme already, while existing events will be migrated in 4.0.

I will release this in 3.6 sometime in the near future.

The expense is here https://opencollective.com/pywebview/expenses/62531
Just submit this amount to the collective. Thanks for the support!

**Comment by BillBridge** [comment]
> @BillBridge PR is here #842 . I ran into an unexpected problem, namely that there is already a `minimized` property on the window object, so minimized as the event name could not be used. After much thought, I decided to migrate events to a new naming convention: `on_minimized`, `on_maximized`, `on_restored`. New events follow this naming scheme already, while existing events will be migrated in 4.0.
> 
> I will release this in 3.6 sometime in the near future.
> 
> The expense is here https://opencollective.com/pywebview/expenses/62531 Just submit this amount to the collective. Thanks for the support!

Hi Ronan,
Thanks for implementing this. I have submitted the amount to the collective. Received in good order?

**Comment by r0x0r** [comment]
Got it. Thanks for the support!

**Comment by BillBridge** [comment]
> Got it. Thanks for the support!

Hi Roman,

Another request: could you also add an event: on_resize?
When the enduser resizes the pywebview window this event shall be triggered.

And do you have suggestions for other events as well of which you may think these will be of added value to pywebview?
Of course I will support this financially.

Thanks in advance for your quick reply.
Warm regards,
Wim
